src/sqlfluff/config.py

```diff
@@ -265,9 +265,29 @@ def load_config_up_to_path(self, path):
         """Loads a selection of config files from both the path and it's parent paths."""
         user_appdir_config = self.load_user_appdir_config()
         user_config = self.load_user_config()
+        config_paths = self.find_config_up_to_path(path)
+        config_stack = [self.load_config_at_path(p) for p in config_paths]
+        return nested_combine(user_appdir_config, user_config, *config_stack)
+
+    def find_ignore_config_files(
+        self, path, working_path=os.getcwd(), ignore_file_name=".sqlfluffignore"
+    ):
+        """Finds sqlfluff ignore files from both the path and its parent paths."""
+        return set(
+            filter(
+                os.path.isfile,
+                map(
+                    lambda x: os.path.join(x, ignore_file_name),
+                    self.find_config_up_to_path(path=path, working_path=working_path),
+                ),
+            )
+        )
 
-        working_path = os.getcwd()
+    def find_config_up_to_path(self, path, working_path=os.getcwd()):
+        """Finds files from both the path and it's parent paths."""

```

Comment:
Perhaps a nit, but I think it's important.

Can we rename this method? Reading the logic, it doesn't really find configs, it does something more like finding the paths to search for configs in, given some parameters.

`find_config_locations_up_to_path` or `find_config_paths_up_to_path` could be good names. I think the name  and docstring are a bit msileading at the moment.

*Bonus points*: I think the actual way thing method is working right now it's actually more like a generator, so perhaps we should actually call is `iter_config_paths_to_path`, and instead of building up our `config_stack`, we actually just `yield path` at each point in this method. It could be a much neater way of expressing the same logic.

Reply:
So I refactored to use `yield` but I had to make additional changes to get the same behavior.

In the existing logic we were visiting the given path and going up parent by parent. On each visit we would insert the path to the top of the stack. If I had just replaced with yield, we would be returning the paths in a reversed order compared to the previous logic (which matters for config overwriting priorities, I verified and it broke a test).

So I made some changes to start from the common path and go down the paths instead.